README.md

```diff
@@ -42,7 +42,346 @@ Or install it yourself as:
+  #
+  namespace 'MyApp::Mailer'
+```
+
+All those global configurations can be overwritten at a finer grained level:
+`mailers`. Each `mailer` has its own copy of the global configuration, so
+that changes are inherited from the top to the bottom, but not bubbled up in the
+opposite direction.
+
+```ruby
+require 'lotus/mailer'
+
+Lotus::Mailer.configure do
+  root '/path/to/root'
+end
+
+class Show
+  include Lotus::Mailer
+  root '/another/root'
+end
+
+Lotus::Mailer.configuration.root # => #<Pathname:/path/to/root>
+Show.root                      # => #<Pathname:/another/root>
+```
+
+### Delivery Details
+
+When a Ruby object includes __Lotus::Mailer__, a developer can customize delivery details, such as `from`, `to` and `subject`.
+
+The `from` field determines the sender of the email. It can accept a string:
+
+```ruby
+class InvoiceMailer
+  include Lotus::Mailer
+
+  from "noreply@example.com"
+end
+```
+
+The `from` field can also accept a `Proc` to determine its value:
+
+```ruby
+class InvoiceMailer
+  include Lotus::Mailer
+
+  from -> { customized_sender }
+
+  def customized_sender
+    "user-#{ user.id }@example.com"
+  end
+end
+```
+
+The `to` field represents one or more emails that will be the recipients of the mail.
+
+It can accept a string:
+
+```ruby
+class InvoiceMailer
+  include Lotus::Mailer
+
+  to "noreply@example.com"
+end
+```
+
+The to field also accept an array of strings:
+
+```ruby
+class InvoiceMailer
+  include Lotus::Mailer
+
+  to "[noreply1@example.com, noreply2@example.com]"
+end
+```
+
+`To` field can also be configured using a `Proc` to determine its value:
+
+```ruby
+class InvoiceMailer
+  include Lotus::Mailer
+
+  to -> { customized_recipient }
+
+  def customized_recipient
+    "user-#{ user.id }@example.com"
+  end
+end
+```
+
+The subject of the message is determined by the field `subject`. It can accept a string:
+
+```ruby
+class InvoiceMailer
+  include Lotus::Mailer
+
+  subject "This is the subject"
+end
+```
+
+The `subject` field can also accept a `Proc` to determine its value:
+
+```ruby
+class InvoiceMailer
+  include Lotus::Mailer
+
+  subject -> { customized_subject }
+
+  def customized_subject
+    "This is the subject"
+  end
+end
+```
+
+### Delivery method
+
+The global delivery method is defined through the __Lotus::Mailer__ configuration, as:
+
+```ruby
+Lotus::Mailer.configuration do
+  delivery_method :smtp
+end
+```
+
+```ruby
+Lotus::Mailer.configuration do
+  delivery_method :smtp, address: "localhost", port:1025
+end
+```
+
+or using the `delivery` alias:
+
+```ruby
+Lotus::Mailer.configuration do
+  delivery :smtp
+end
+```
+
+It is also possible to configure a custom delivery method with a variable:
+
+```ruby
+MyCustomDeliveryMethod = :smtp
+
+Lotus::Mailer.configure do
+  delivery MyCustomDeliveryMethod, foo:'bar'
+end
+```
+
+The delivery method specified must be compatible with the `Mail` gem, and all `Mail` gem delivery methods are available in __Lotus::Mailer__.
```

Comment:
@joneslee85 Maybe the link would suffice, since there are a lot of little details in the Mail gem.